x-pack/plugins/security_solution/cypress/tasks/alerts_detection_rules.ts

```diff
@@ -360,16 +363,73 @@ export const importRules = (rulesFile: string) => {
   cy.get(INPUT_FILE).should('not.exist');
 };
 
-export const expectNumberOfRules = (expectedNumber: number) => {
-  cy.get(RULES_TABLE).then(($table) => {
-    const rulesRow = cy.wrap($table.find(RULES_ROW));
+export const expectRulesManagementTab = () =>
+  cy.get(RULES_MANAGEMENT_TAB).should('have.class', 'euiTab-isSelected');
+export const expectRulesMonitoringTab = () =>
+  cy.get(RULES_MONITORING_TAB).should('have.class', 'euiTab-isSelected');
 
-    rulesRow.should('have.length', expectedNumber);
-  });
+export const expectFilterSearchTerm = (searchTerm: string) =>
+  cy.get(RULE_SEARCH_FIELD).should('have.value', searchTerm);
+
+export const expectTags = (tags: string[]) => {
+  cy.get(RULES_TAGS_FILTER_BTN).contains(`Tags${tags.length}`).click();
+
+  cy.get(RULES_TAGS_FILTER_POPOVER)
+    .find(RULES_SELECTED_TAG)
+    .should('have.length', tags.length)
+    .each(($el, index) => {
+      cy.wrap($el).contains(tags[index]);
+    });
+
+  for (const tag of tags) {
+    cy.get(RULES_TAGS_FILTER_POPOVER).contains(RULES_SELECTED_TAG, tag);
+  }
```

Comment:
You are right `for .. of` is basically a duplication. The only difference is that `for` loop doesn't regard tags ordering while the section above requires `tags` to be sorted alphabetically.

I removed `for` loop section since it doesn't bring anything here.